platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -1729,13 +1729,15 @@ public InternalProblems getProblems() {
     private void assertNotDetachedExtensionDoingExtending(Iterable<Configuration> extendsFrom) {
         if (isDetachedConfiguration()) {
             String summarizedExtensionTargets = StreamSupport.stream(extendsFrom.spliterator(), false)
-                .map(c -> "'" + c.getName() + "'")
+                .map(ConfigurationInternal.class::cast)
+                .map(ConfigurationInternal::getDisplayName)
                 .collect(Collectors.joining(", "));
-            DeprecationLogger.deprecateAction(String.format("Calling extendsFrom on %s", this.getDisplayName()))
-                .withContext(String.format("Detached configurations should not extend other configurations, this was extending: %s.", summarizedExtensionTargets))
-                .willBecomeAnErrorInGradle9()
-                .withUpgradeGuideSection(8, "detached_configurations_cannot_extend")
-                .nagUser();
+            GradleException ex = new GradleException(getDisplayName() + " cannot extend " + summarizedExtensionTargets);
+            ProblemId id = ProblemId.create("extend-detached-not-allowed", "Extending a detachedConfiguration is not allowed", GradleCoreProblemGroup.configurationUsage());
```

Comment:
```suggestion
            ProblemId id = ProblemId.create("extend-detached-not-allowed", "Extending a detachedConfiguration is not allowed.", GradleCoreProblemGroup.configurationUsage());
```

💭 Looking at the error message in the test, this feels like it needs a period

Reply:
I'm going to leave this as is for now, as it seems to be in line with other problems we create via the Problems API.  We should think about consistency in error message formatting at some point though.